### PR TITLE
new raycasts

### PR DESCRIPTION
**Pull request by nearnshaw**

### CODE REVIEW

File: content/creator/sdk7/interactivity/raycasting.md [hunk 1]
```diff
@@ -14,79 +14,98 @@ Raycasting is a fundamental tool in game development. With raycasting, you can t
...
+A Raycast component describes the invisible ray that is used to query for intersecting entities. The ray is traced starting at the entity's position, as defined by the Transform component and affected by that of any parent entities. The direction can be defined in various ways, 
 
-- `origin`: _Vector3_ with the coordinates in scene space to start the ray from. 
-- `direction`: _Vector3_ describing the direction of the ray (as if the ray started from _0,0,0_).
+Rays are defined using the following data:
+
+
+- `direction`: _Vector3_ or _Entity_ describing the direction of the ray. Its behavior will depend on the mode of the ray:
+	- `LOCAL_DIRECTION`: A direction relative to the forward-facing direction of the entity, affected also by the transformation of any parent entities.
+	- `GLOBAL_DIRECTION`: Ignores the entity's rotation, and faces a direction as if the entity's rotation was 0.
```
Comment (menduz) [inline content/creator/sdk7/interactivity/raycasting.md]:
```suggestion
	- `GLOBAL_DIRECTION`: Ignores the entity's rotation, and faces a direction as if the entity's rotation was 0. This is useful to i.e. always point down.
```

File: content/creator/sdk7/interactivity/raycasting.md [hunk 2]
```diff
@@ -14,79 +14,98 @@ Raycasting is a fundamental tool in game development. With raycasting, you can t
...
-A Raycast component describes the invisible ray that will be used to query for entities. Rays are defined using the following data:
+A Raycast component describes the invisible ray that is used to query for intersecting entities. The ray is traced starting at the entity's position, as defined by the Transform component and affected by that of any parent entities. The direction can be defined in various ways, 
 
-- `origin`: _Vector3_ with the coordinates in scene space to start the ray from. 
-- `direction`: _Vector3_ describing the direction of the ray (as if the ray started from _0,0,0_).
+Rays are defined using the following data:
+
+
+- `direction`: _Vector3_ or _Entity_ describing the direction of the ray. Its behavior will depend on the mode of the ray:
+	- `LOCAL_DIRECTION`: A direction relative to the forward-facing direction of the entity, affected also by the transformation of any parent entities.
```
Comment (menduz) [inline content/creator/sdk7/interactivity/raycasting.md]:
```suggestion
	- `LOCAL_DIRECTION`: A direction relative to the forward-facing direction of the entity, affected also by the transformation of any parent entities. This is useful to detect obstacles in front of vehicles honoring their heading.
```

File: content/creator/sdk7/interactivity/raycasting.md [hunk 3]
```diff
@@ -14,79 +14,98 @@ Raycasting is a fundamental tool in game development. With raycasting, you can t
...
+	- `GLOBAL_DIRECTION`: Ignores the entity's rotation, and faces a direction as if the entity's rotation was 0.
+	- `GLOBAL_TARGET`: Traces a line between the entity's position and a target global position in the scene. It ignores the entity's rotation.
```
Comment (menduz) [inline content/creator/sdk7/interactivity/raycasting.md]:
```suggestion
	- `GLOBAL_TARGET`: Traces a line between the entity's position and a target global position in the scene. It ignores the entity's rotation. Useful to create tower defense games, each tower's turret can point to a pin-pointed coordinate in space.
```

File: content/creator/sdk7/interactivity/raycasting.md [hunk 4]
```diff
@@ -14,80 +14,139 @@ Raycasting is a fundamental tool in game development. With raycasting, you can t
...
+
+Transform.create(entity1, {
+	position: Vector3.create(8, 1, 0)
+})
 
-// only return first entity
-Raycast.createOrReplace(engine.addEntity(), {
-  origin: Vector3.create(8, 1, 0),
-  direction: Vector3.create(0, 0, 1),
++Raycast.createOrReplace(entity1, {
```
Comment (pravusjif) [inline content/creator/sdk7/interactivity/raycasting.md:55]: mini typo in this line (the initial `+` char)

Comment (pravusjif) [inline content/creator/sdk7/interactivity/raycasting.md]:
```suggestion
Raycast.createOrReplace(entity1, {
```

**Review by pravusjif — APPROVED** [review]

**Pull request merged**